packages/grafana-e2e/src/flows/addPanel.ts

```diff
@@ -2,27 +2,57 @@ import { e2e } from '../index';
 import { getScenarioContext } from '../support/scenarioContext';
 
 export interface AddPanelConfig {
+  dashboardUid?: string;
   dataSourceName: string;
   queriesForm: Function;
+  visualizationName: string;
 }
 
 const DEFAULT_ADD_PANEL_CONFIG: AddPanelConfig = {
   dataSourceName: 'TestData DB',
   queriesForm: () => {},
+  visualizationName: 'Graph',
 };
 
-export const addPanel = (config?: Partial<AddPanelConfig>) => {
-  const { dataSourceName, queriesForm } = { ...DEFAULT_ADD_PANEL_CONFIG, ...config };
-
-  getScenarioContext().then(({ lastAddedDashboardUid }: any) => {
-    e2e.flows.openDashboard(lastAddedDashboardUid);
-    e2e.pages.Dashboard.Toolbar.toolbarItems('Add panel').click();
-    e2e.pages.AddDashboard.addNewPanel().click();
-    e2e()
-      .get('.ds-picker')
-      .click()
-      .contains('[id^="react-select-"][id*="-option-"]', dataSourceName)
-      .click();
-    queriesForm();
-  });
+// @todo this actually returns type `Cypress.Chainable`
+export const addPanel = (config?: Partial<AddPanelConfig>): any => {
+  const { dashboardUid, dataSourceName, queriesForm, visualizationName } = { ...DEFAULT_ADD_PANEL_CONFIG, ...config };
+  const panelTitle = `e2e-${Date.now()}`;
+
+  return getScenarioContext()
+    .then(({ lastAddedDashboardUid }: any) => {
+      e2e.flows.openDashboard(dashboardUid ?? lastAddedDashboardUid);
+      e2e.pages.Dashboard.Toolbar.toolbarItems('Add panel').click();
+      e2e.pages.AddDashboard.addNewPanel().click();
+
+      e2e()
+        .get('.ds-picker')
+        .click()
+        .contains('[id^="react-select-"][id*="-option-"]', dataSourceName)
```

Comment:
Maybe consider adding an aria-label selector instead of using `[id^="react-select-"][id*="-option-"]`?

Reply:
Yup, I agree.